core/src/main/java/com/oath/oak/OakDetachedReadValueBuffer.java

```diff
@@ -6,232 +6,25 @@
-     *
-     * @param transformer the function that executes the transformation
-     * @return a transformation of the ByteBuffer content
-     * @throws NullPointerException if the transformer is null
-     */
-    public <T> T transform(OakTransformer<T> transformer) {
-        if (transformer == null) {
-            throw new NullPointerException();
-        }
-        // Use a "for" loop to ensure maximal retries.
-        for (int i = 0; i < 1024; i++) {
-            Result<T> result = valueOperator.transform(getValueSlice(), transformer, version);
-            if (result.operationResult == FALSE) {
-                throw new ConcurrentModificationException();
-            } else if (result.operationResult == RETRY) {
-                lookupValueReference();
-                continue;
-            }
-            return result.value;
-        }
-
-        throw new RuntimeException("Transform failed: reached retry limit (1024).");
-    }
-
-    private Slice start(Slice valueSlice) {
-        ValueUtils.ValueResult res = valueOperator.lockRead(valueSlice, version);
-        if (res == FALSE) {
-            throw new ConcurrentModificationException();
-        }
-        // In case the value moved or was the version does not match
-        if (res == RETRY) {
-            lookupValueReference();
-            return start(getValueSlice());
-        }
-        return valueSlice;
-    }
-
-    private void end(Slice valueSlice) {
-        valueOperator.unlockRead(valueSlice, version);
-    }
-
-    private void lookupValueReference() {
-        EntrySet.LookUp lookUp = internalOakMap.refreshValuePosition(keyReference);
-        if (lookUp == null || lookUp.valueSlice == null) {
-            throw new ConcurrentModificationException();
-        }
-        valueReference = lookUp.valueReference;
-        version = lookUp.version;
-    }
-
-    /*-------------- OakUnsafeRef --------------*/
-
-    @Override
-    public ByteBuffer getByteBuffer() {
-        ByteBuffer buff = getValueSlice().getByteBuffer().asReadOnlyBuffer();
-        int position = valuePosition();
-        int limit = position + capacity();
-        buff.position(position);
-        buff.limit(limit);
-        return buff.slice();
-    }
-
-    @Override
-    public int getOffset() {
-        return 0;
+    OakDetachedReadValueBuffer(int headerSize) {
+        super();
+        this.value = new EntrySet.ValueBuffer(headerSize);
     }
 
     @Override
-    public int getLength() {
-        return capacity();
+    protected OakAttachedReadBuffer getAttachedBuffer() {
+        return value;
     }
 
     @Override
-    public long getAddress() {
-        ByteBuffer buff = getValueSlice().getByteBuffer();
-        long address = ((DirectBuffer) buff).address();
-        return address + valuePosition();
+    protected <T> T transformBuffer(Function<OakAttachedReadBuffer, T> transformer) {
+        // Internal call. No input validation.
+        return transformer.apply(value);
```

Comment:
Again I do not understand why no buffer rebuild here...

Reply:
See the comment above.